bot/processors/pennychat.py

```diff
@@ -0,0 +1,289 @@
+                    "action_id": "penny_chat_channel_select",
+                }
+            },
+        ]
+    }
+
+    return template
+
+
+class PennyChatBotModule(BotModule):
+    def __init__(self, slack):
+        self.slack = slack
+
+    @classmethod
+    def create_penny_chat(cls, slack, event):
+        response = slack.chat_postMessage(channel=event['channel_id'],
+                                          user=event['user_id'],
+                                          blocks=create_penny_chat_blocks()
+                                          )
+
+        PennyChat.objects.create(template_channel=response.data['channel'], template_timestamp=response.data['ts'])
+
+    @is_block_interaction_event
+    @is_action_id('penny_chat_user_select')
+    def user_select(self, event):
+        users = event['actions'][0]['selected_users']
+        penny_chat = PennyChat.objects.get(view=event['view']['id'])
+        invitees = ''
+        for invitee in users:
+            invitees = invitees + ',' + invitee if len(invitees) > 0 else invitee
+        penny_chat.invitees = invitees
+        penny_chat.save()
+
+    @is_block_interaction_event
+    @is_action_id('penny_chat_channel_select')
+    def channel_select(self, event):
+        selected_channels = event['actions'][0]['selected_channels']
+        penny_chat = PennyChat.objects.get(view=event['view']['id'])
+        channels = ''
+        for channel in selected_channels:
+            channels = channel + ',' + channel if len(selected_channels) > 0 else channel
+        penny_chat.channels = channels
+        penny_chat.save()
+
+    @is_block_interaction_event
+    @is_action_id('penny_chat_time')
+    def time_select(self, event):
+        time = event['actions'][0]['selected_option']['value']
+        penny_chat = PennyChat.objects.get(view=event['view']['id'])
+        date = str(penny_chat.date.date())
+        penny_chat.date = datetime.strptime(date + ' ' + time, '%Y-%m-%d %I:%M %p')
+        penny_chat.save()
+
+    @is_block_interaction_event
+    @is_action_id('penny_chat_date')
+    def date_select(self, event):
+        date = event['actions'][0]['selected_date']
+        penny_chat = PennyChat.objects.get(view=event['view']['id'])
+        time = str(penny_chat.date.time())
+        penny_chat.date = datetime.strptime(date + ' ' + time, '%Y-%m-%d %H:%M:%S')
+        penny_chat.save()
+
+    @is_block_interaction_event
+    @is_action_id('penny_chat_details')
+    def open_details_view(self, event):
+        penny_chat = PennyChat.objects.get(template_timestamp=event['container']['message_ts'])
+        modal = penny_chat_details_modal(penny_chat)
+        response = self.slack.views_open(view=modal, trigger_id=event['trigger_id'])
+        penny_chat.view = response.data['view']['id']
+        penny_chat.save()
+
+    @is_event_type('view_submission')
+    @has_callback_id('penny_chat_details')
+    def submit_details(self, event):
```

Comment:
Would it simplify the implementation if we just created the `PennyChat` object with the data from the submitted view? If the user abandons the flow half way through, we will have a partially created penny chat object.